### PR TITLE
Create OpenAPIV3 schema for vsphere-csi package and generate package with the schema

### LINKED ISSUE #2831: Create OpenAPIV3 schema for vsphere-csi in TCE
## Feature Request

Create OpenAPIV3 schema for vsphere-csi in TCE

### PR DESCRIPTION
**Pull request by blc1996**
Signed-off-by: Lucheng Bao <luchengb@vmware.com>

## What this PR does / why we need it

Create OpenAPIV3 schema for vsphere-csi package and generate package with the schema

## Details for the Release Notes (PLEASE PROVIDE)

```release-note
Create OpenAPIV3 schema for vsphere-csi package and generate package with the schema
```

## Which issue(s) this PR fixes

Fixes: #2831 

## Describe testing done for PR

Tested with ytt commands and ran make generate-openapischema-package locally

## Special notes for your reviewer

YTT version needs to be bumped to latest (0.38.0)

### CODE REVIEW

**Comment by dvonthenen** [comment]
@vijaykatam I think we are looking for a review on this

**Review by vijaykatam — DISMISSED** [review]

**Comment by seemiller** [comment]
@vijaykatam @blc1996 Sorry about this, be we got a merge conflict. Could you please resolve?

**Comment by seemiller** [comment]
@vijaykatam @blc1996 ping - can someone merge these 2 files?

**Comment by blc1996** [comment]
Hi @seemiller , I'm not sure if we still want to merge this. Let me sync up with @peterochodo as he is not taking care of CSI package.

**Comment by blc1996** [comment]
> Hi @seemiller , I'm not sure if we still want to merge this. Let me sync up with @peterochodo as he is not taking care of CSI package.

Looks like Peter is creating another PV CSI package. I'll resolve the conflict and get it merged.

**Comment by blc1996** [comment]
Hi @vijaykatam, could you take a look at this PR again? Thanks!

**Comment by stmcginnis** [comment]
Looks like this has a merge conflict again. Is this still needed?

File: addons/packages/vsphere-csi/2.3.0/package.yaml [hunk 1]
```diff
@@ -5,21 +5,106 @@ metadata:
...
   licenses:
-    - "Apache 2.0"
+  - Apache 2.0
   template:
     spec:
       fetch:
-        - imgpkgBundle:
-            image: projects.registry.vmware.com/tce/vsphere-csi@sha256:a2814538e00b6a875ead5cbc3bfd43335b5fa981c2d6c5ead42058eab79edf4e
+      - imgpkgBundle:
+          image: projects.registry.vmware.com/tce/vsphere-csi@sha256:069990d21488f6ffea7fe5cc33e074297294a52f3f202fbe346745e8fbc55aef
```
Comment (seemiller) [inline addons/packages/vsphere-csi/2.3.0/package.yaml:15]:
```suggestion
          image: projects.registry.vmware.com/tce/vsphere-csi@sha256:337a51c8d72de40e1815eb56d98cb6ee0912f2fa7ac391f4f9711d910ece7506
```

File: addons/packages/vsphere-csi/2.4.1/package.yaml [hunk 2]
```diff
@@ -5,21 +5,118 @@ metadata:
...
   licenses:
-    - "Apache 2.0"
+  - Apache 2.0
   template:
     spec:
       fetch:
-        - imgpkgBundle:
-            image: projects.registry.vmware.com/tce/vsphere-csi@sha256:c615f4a95de0161f244f644cfb512449ef8b2c233e63541d16fa44bcd7bd28b6
+      - imgpkgBundle:
+          image: projects.registry.vmware.com/tce/vsphere-csi@sha256:ffb6731278dc157d75ff56e8a4e8f6b5f8f08cc108a86e42c0cb8204ae7203ef
```
Comment (seemiller) [inline addons/packages/vsphere-csi/2.4.1/package.yaml:15]:
```suggestion
          image: projects.registry.vmware.com/tce/vsphere-csi@sha256:ed2b01baf6e93a5a7bfb7d79f3602735ab4cd3ab0e3a68c9856a64e6894a3ae3
```
Locally I rebased with main and pushed the package, this is the SHA that I got.

**Review by seemiller — CHANGES_REQUESTED** [review]

**Comment by seemiller** [comment]
If this PR is still needed, I would recommend rebasing with the main branch, pushing the packages and verifying the SHAs in the package.yaml files.

**Comment by stmcginnis** [comment]
Can this be closed?

**Comment by blc1996** [comment]
Closing this MR. If this is needed, please open it against main branch once more @vijaykatam . Thanks!

**Pull request closed without merging**